src/cascadia/TerminalSettingsEditor/ColorSchemesPageViewModel.cpp

```diff
@@ -52,6 +52,33 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
             }
             it.MoveNext();
         }
+        if (!it.HasCurrent())
+        {
+            // we didn't find the previously selected scheme, just select the first one
+            CurrentScheme(_AllColorSchemes.GetAt(0));

```

Comment:
FYI, this will not trigger a cascading update on `CanDeleteCurrentScheme`. If we select an inbox scheme, the user will be able to delete it.

You can get rid of `ColorSchemeSelectionChanged` by binding `CurrentItem` to `CurrentScheme` (TwoWay) and then adding a local property changed handler from CurrentScheme to fire _another_ property changed event for `CanDelete...`

Or, you can skip the second part _and implement CurrentScheme as an observable property yourself!_ In cases like this, it can be worth avoiding the easy all-in-one macro... :smile:

Reply:
That makes sense! Fixed